### PR TITLE
python3Packages.gym: Remove stale substituteInPlace, pythonPackages.*: Disable tests, treewide: update permanent url redirects

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/qreactor/default.nix [hunk 1]
```diff
@@ -24,14 +23,13 @@ buildPythonPackage rec {
...
-  checkInputs = [
-    pyqt5
-  ];
-
   pythonImportsCheck = [
     "qreactor"
   ];
 
+  # no tests executed
+  doCheck = false;
```
Comment (raboof) [inline pkgs/development/python-modules/qreactor/default.nix:31]: Why remove this? While there are no explicit tests, it does test compiling the code works, right?

Comment (SuperSandro2000) [reply pkgs/development/python-modules/qreactor/default.nix:31]: > Why remove this?

I greped for ``Ran 0 tests in 0.000s`` in the logs and disabled tests in all matching files.

> While there are no explicit tests, it does test compiling the code works, right?

There is a pythonImportsCheck which is enough.

Comment (raboof) [reply pkgs/development/python-modules/qreactor/default.nix:31]: > There is a pythonImportsCheck which is enough.

If it's equivalent, why does `pythonImportsCheck` not need the pyqt5 input?

Comment (raboof) [reply pkgs/development/python-modules/qreactor/default.nix:31]: Also I'm still not sure what the advantage of explicitly disabling checks is: it just means when upstream ever starts adding tests we're likely to miss them.

Anyway, even if I don't understand the motivation, I'm not opposed to the change. Perhaps at least say something about it in the commit message?

Comment (SuperSandro2000) [reply pkgs/development/python-modules/qreactor/default.nix:31]: > If it's equivalent, why does pythonImportsCheck not need the pyqt5 input?

It takes propagatedBuildInputs and does something like ``python -c 'import X'``

> Also I'm still not sure what the advantage of explicitly disabling checks is

We do not surprisingly execute tests when updating.

> it just means when upstream ever starts adding tests we're likely to miss them.

That is the duty of the maintainer to check. For Python the importsCheck is usually enough and catches the same bugs as running the tests itself for small packages. Most other test failures are logic mistakes.

> Perhaps at least say something about it in the commit message?

This is the 5th PR of this kind and to be honest I am to lazy to do a proper description.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110824` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages marked as broken and skipped:
</summary>
<ul>
<li>python37Packages.rl-coach</li>
<li>python38Packages.rl-coach</li>
<li>python39Packages.baselines</li>
</ul>
</details>
<details>

  <summary>5 packages failed to build and are new build failures:</summary>
  <ul>
<li>inkcut: log was empty</li>
<li>inkscape-extensions.inkcut: log was empty</li>
<li>python37Packages.qreactor: log <a href="https://termbin.com/gzzs">https://termbin.com/gzzs</a></li>
<li>python38Packages.qreactor: log <a href="https://termbin.com/j7rs">https://termbin.com/j7rs</a></li>
<li>python39Packages.qreactor: log <a href="https://termbin.com/m31h">https://termbin.com/m31h</a></li>
  </ul>
</details>
<details>
<summary>
37 packages built:
</summary>
<ul>
<li>coloredlogs (python38Packages.coloredlogs)</li>
<li>executor (python38Packages.executor)</li>
<li>ocrmypdf</li>
<li>python37Packages.baselines</li>
<li>python37Packages.capturer</li>
<li>python37Packages.coloredlogs</li>
<li>python37Packages.executor</li>
<li>python37Packages.gym</li>
<li>python37Packages.property-manager</li>
<li>python37Packages.qimage2ndarray</li>
<li>python37Packages.roboschool</li>
<li>python37Packages.rotate-backups</li>
<li>python37Packages.stytra</li>
<li>python37Packages.update-dotdee</li>
<li>python37Packages.zigpy-znp</li>
<li>python38Packages.baselines</li>
<li>python38Packages.capturer</li>
<li>python38Packages.gym</li>
<li>python38Packages.property-manager</li>
<li>python38Packages.qimage2ndarray</li>
<li>python38Packages.roboschool</li>
<li>rotate-backups (python38Packages.rotate-backups)</li>
<li>python38Packages.stytra</li>
<li>update-dotdee (python38Packages.update-dotdee)</li>
<li>python38Packages.zigpy-znp</li>
<li>python39Packages.capturer</li>
<li>python39Packages.coloredlogs</li>
<li>python39Packages.executor</li>
<li>python39Packages.gym</li>
<li>python39Packages.property-manager</li>
<li>python39Packages.qimage2ndarray</li>
<li>python39Packages.roboschool</li>
<li>python39Packages.rotate-backups</li>
<li>python39Packages.update-dotdee</li>
<li>python39Packages.zigpy-znp</li>
<li>qnotero</li>
<li>rofimoji</li>
</ul>
</details>

File: pkgs/development/python-modules/qreactor/default.nix (hunk 1 above)
Comment (raboof) [reply pkgs/development/python-modules/qreactor/default.nix:31]: > We do not surprisingly execute tests when updating.

Surprisingly *not* executing tests seems worse to me...

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110824` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages marked as broken and skipped:
</summary>
<ul>
<li>python37Packages.rl-coach</li>
<li>python38Packages.rl-coach</li>
<li>python39Packages.baselines</li>
</ul>
</details>
<details>
<summary>
37 packages built:
</summary>
<ul>
<li>coloredlogs (python38Packages.coloredlogs)</li>
<li>executor (python38Packages.executor)</li>
<li>ocrmypdf</li>
<li>python37Packages.baselines</li>
<li>python37Packages.capturer</li>
<li>python37Packages.coloredlogs</li>
<li>python37Packages.executor</li>
<li>python37Packages.gym</li>
<li>python37Packages.property-manager</li>
<li>python37Packages.qimage2ndarray</li>
<li>python37Packages.roboschool</li>
<li>python37Packages.rotate-backups</li>
<li>python37Packages.stytra</li>
<li>python37Packages.update-dotdee</li>
<li>python37Packages.zigpy-znp</li>
<li>python38Packages.baselines</li>
<li>python38Packages.capturer</li>
<li>python38Packages.gym</li>
<li>python38Packages.property-manager</li>
<li>python38Packages.qimage2ndarray</li>
<li>python38Packages.roboschool</li>
<li>rotate-backups (python38Packages.rotate-backups)</li>
<li>python38Packages.stytra</li>
<li>update-dotdee (python38Packages.update-dotdee)</li>
<li>python38Packages.zigpy-znp</li>
<li>python39Packages.capturer</li>
<li>python39Packages.coloredlogs</li>
<li>python39Packages.executor</li>
<li>python39Packages.gym</li>
<li>python39Packages.property-manager</li>
<li>python39Packages.qimage2ndarray</li>
<li>python39Packages.roboschool</li>
<li>python39Packages.rotate-backups</li>
<li>python39Packages.update-dotdee</li>
<li>python39Packages.zigpy-znp</li>
<li>qnotero</li>
<li>rofimoji</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110824` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages marked as broken and skipped:
</summary>
<ul>
<li>python37Packages.rl-coach</li>
<li>python38Packages.rl-coach</li>
<li>python39Packages.baselines</li>
</ul>
</details>
<details>
<summary>
37 packages built:
</summary>
<ul>
<li>coloredlogs (python38Packages.coloredlogs)</li>
<li>executor (python38Packages.executor)</li>
<li>ocrmypdf</li>
<li>python37Packages.baselines</li>
<li>python37Packages.capturer</li>
<li>python37Packages.coloredlogs</li>
<li>python37Packages.executor</li>
<li>python37Packages.gym</li>
<li>python37Packages.property-manager</li>
<li>python37Packages.qimage2ndarray</li>
<li>python37Packages.roboschool</li>
<li>python37Packages.rotate-backups</li>
<li>python37Packages.stytra</li>
<li>python37Packages.update-dotdee</li>
<li>python37Packages.zigpy-znp</li>
<li>python38Packages.baselines</li>
<li>python38Packages.capturer</li>
<li>python38Packages.gym</li>
<li>python38Packages.property-manager</li>
<li>python38Packages.qimage2ndarray</li>
<li>python38Packages.roboschool</li>
<li>rotate-backups (python38Packages.rotate-backups)</li>
<li>python38Packages.stytra</li>
<li>update-dotdee (python38Packages.update-dotdee)</li>
<li>python38Packages.zigpy-znp</li>
<li>python39Packages.capturer</li>
<li>python39Packages.coloredlogs</li>
<li>python39Packages.executor</li>
<li>python39Packages.gym</li>
<li>python39Packages.property-manager</li>
<li>python39Packages.qimage2ndarray</li>
<li>python39Packages.roboschool</li>
<li>python39Packages.rotate-backups</li>
<li>python39Packages.update-dotdee</li>
<li>python39Packages.zigpy-znp</li>
<li>qnotero</li>
<li>rofimoji</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110824` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
11 packages marked as broken and skipped:
</summary>
<ul>
<li>python37Packages.baselines</li>
<li>python37Packages.gym</li>
<li>python37Packages.rl-coach</li>
<li>python37Packages.roboschool</li>
<li>python38Packages.baselines</li>
<li>python38Packages.gym</li>
<li>python38Packages.rl-coach</li>
<li>python38Packages.roboschool</li>
<li>python39Packages.baselines</li>
<li>python39Packages.gym</li>
<li>python39Packages.roboschool</li>
</ul>
</details>
<details>

  <summary>10 packages failed to build and are new build failures:</summary>
  <ul>
<li>python37Packages.coloredlogs: log <a href="https://termbin.com/qst3">https://termbin.com/qst3</a></li>
<li>python37Packages.executor: log was empty</li>
<li>python37Packages.property-manager: log was empty</li>
<li>python37Packages.rotate-backups: log was empty</li>
<li>python37Packages.update-dotdee: log was empty</li>
<li>python39Packages.coloredlogs: log <a href="https://termbin.com/54dp">https://termbin.com/54dp</a></li>
<li>python39Packages.executor: log was empty</li>
<li>python39Packages.property-manager: log was empty</li>
<li>python39Packages.rotate-backups: log was empty</li>
<li>python39Packages.update-dotdee: log was empty</li>
  </ul>
</details>
<details>
<summary>
12 packages built:
</summary>
<ul>
<li>python37Packages.capturer</li>
<li>python37Packages.pyres</li>
<li>python37Packages.qimage2ndarray</li>
<li>python37Packages.stytra</li>
<li>python38Packages.capturer</li>
<li>python38Packages.pyres</li>
<li>python38Packages.qimage2ndarray</li>
<li>python38Packages.stytra</li>
<li>python39Packages.capturer</li>
<li>python39Packages.pyres</li>
<li>python39Packages.qimage2ndarray</li>
<li>qnotero</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110824` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
22 packages built:
</summary>
<ul>
<li>python37Packages.capturer</li>
<li>python37Packages.coloredlogs</li>
<li>python37Packages.executor</li>
<li>python37Packages.property-manager</li>
<li>python37Packages.pyres</li>
<li>python37Packages.qimage2ndarray</li>
<li>python37Packages.rotate-backups</li>
<li>python37Packages.stytra</li>
<li>python37Packages.update-dotdee</li>
<li>python38Packages.capturer</li>
<li>python38Packages.pyres</li>
<li>python38Packages.qimage2ndarray</li>
<li>python38Packages.stytra</li>
<li>python39Packages.capturer</li>
<li>python39Packages.coloredlogs</li>
<li>python39Packages.executor</li>
<li>python39Packages.property-manager</li>
<li>python39Packages.pyres</li>
<li>python39Packages.qimage2ndarray</li>
<li>python39Packages.rotate-backups</li>
<li>python39Packages.update-dotdee</li>
<li>qnotero</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110824` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages marked as broken and skipped:
</summary>
<ul>
<li>python37Packages.rl-coach</li>
<li>python38Packages.rl-coach</li>
<li>python39Packages.baselines</li>
</ul>
</details>
<details>
<summary>
37 packages built:
</summary>
<ul>
<li>coloredlogs (python38Packages.coloredlogs)</li>
<li>executor (python38Packages.executor)</li>
<li>ocrmypdf</li>
<li>python37Packages.baselines</li>
<li>python37Packages.capturer</li>
<li>python37Packages.coloredlogs</li>
<li>python37Packages.executor</li>
<li>python37Packages.gym</li>
<li>python37Packages.property-manager</li>
<li>python37Packages.qimage2ndarray</li>
<li>python37Packages.roboschool</li>
<li>python37Packages.rotate-backups</li>
<li>python37Packages.stytra</li>
<li>python37Packages.update-dotdee</li>
<li>python37Packages.zigpy-znp</li>
<li>python38Packages.baselines</li>
<li>python38Packages.capturer</li>
<li>python38Packages.gym</li>
<li>python38Packages.property-manager</li>
<li>python38Packages.qimage2ndarray</li>
<li>python38Packages.roboschool</li>
<li>rotate-backups (python38Packages.rotate-backups)</li>
<li>python38Packages.stytra</li>
<li>update-dotdee (python38Packages.update-dotdee)</li>
<li>python38Packages.zigpy-znp</li>
<li>python39Packages.capturer</li>
<li>python39Packages.coloredlogs</li>
<li>python39Packages.executor</li>
<li>python39Packages.gym</li>
<li>python39Packages.property-manager</li>
<li>python39Packages.qimage2ndarray</li>
<li>python39Packages.roboschool</li>
<li>python39Packages.rotate-backups</li>
<li>python39Packages.update-dotdee</li>
<li>python39Packages.zigpy-znp</li>
<li>qnotero</li>
<li>rofimoji</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110824` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages marked as broken and skipped:
</summary>
<ul>
<li>python37Packages.rl-coach</li>
<li>python38Packages.rl-coach</li>
<li>python39Packages.baselines</li>
</ul>
</details>
<details>

  <summary>2 packages failed to build and are new build failures:</summary>
  <ul>
<li>python37Packages.baselines: log was empty</li>
<li>python38Packages.baselines: log was empty</li>
  </ul>
</details>
<details>
<summary>
35 packages built:
</summary>
<ul>
<li>coloredlogs (python38Packages.coloredlogs)</li>
<li>executor (python38Packages.executor)</li>
<li>ocrmypdf</li>
<li>python37Packages.capturer</li>
<li>python37Packages.coloredlogs</li>
<li>python37Packages.executor</li>
<li>python37Packages.gym</li>
<li>python37Packages.property-manager</li>
<li>python37Packages.qimage2ndarray</li>
<li>python37Packages.roboschool</li>
<li>python37Packages.rotate-backups</li>
<li>python37Packages.stytra</li>
<li>python37Packages.update-dotdee</li>
<li>python37Packages.zigpy-znp</li>
<li>python38Packages.capturer</li>
<li>python38Packages.gym</li>
<li>python38Packages.property-manager</li>
<li>python38Packages.qimage2ndarray</li>
<li>python38Packages.roboschool</li>
<li>rotate-backups (python38Packages.rotate-backups)</li>
<li>python38Packages.stytra</li>
<li>update-dotdee (python38Packages.update-dotdee)</li>
<li>python38Packages.zigpy-znp</li>
<li>python39Packages.capturer</li>
<li>python39Packages.coloredlogs</li>
<li>python39Packages.executor</li>
<li>python39Packages.gym</li>
<li>python39Packages.property-manager</li>
<li>python39Packages.qimage2ndarray</li>
<li>python39Packages.roboschool</li>
<li>python39Packages.rotate-backups</li>
<li>python39Packages.update-dotdee</li>
<li>python39Packages.zigpy-znp</li>
<li>qnotero</li>
<li>rofimoji</li>
</ul>
</details>

**Pull request merged**